content/blogs/newsletter-102.md

```diff
@@ -0,0 +1,113 @@
+
+Thanks to his work, the hack in `SepAccessCheck` has been removed and security access checks are now operational and the kernel can prevent any unauthorized access from whoever tries to access an object with improper access rights. His future plan for the Security Subsystem is to implement security auditing of all objects and miscellaneous improvements where needed.
+
+## Registry improvements
+For ReactOS to reach Beta phase it has to solve two major milestones -- stability and corruptions. Usually corruptions in ReactOS can manifest in form of filesystem and registry corruptions.
+While filesystem corruptions can be prevented by choosing a more modern filesystem such as BtrFS which is what we currently support, registry corruptions are by themselves a death sentence as ReactOS currently doesn't implement any self-healing mechanisms to recover the system from corrupted registry hives. As a result, the user is forced to do a clean reinstallation of the system.
+Worst case, the system would still boot up but be stuck on the login prompt dialog (aka the "Ctrl+Alt+Del" dialog) as the user profile has been corrupted or missing. The latter generally happens after ReactOS is forcefully shutdown after a clean install.
+
+George Bișoc is currently working on improving the registry by implementing the necessary registry self-healing mechanisms and `CmCheckRegistry`, a fundamental function that validates the registry for any corrupted parts of the registry and takes appropriate actions at recovering a registry hive from such corrupt parts. While his work is currently in progress ([#4571](https://github.com/reactos/reactos/pull/4571)), so far he has received some positive results from testers. A small piece of his work has been merged, which fixes the registry flusher of ReactOS not working for almost a decade! This fix also solves the problem of the missing user profile due to forcefully shutting down the system, and as such you should no longer expect to see the infamous "Ctrl+Alt+Del" dialog prompt as often as before.
+
+In addition he is working on the registry cache implementation ([#5088](https://github.com/reactos/reactos/pull/5088)), as ReactOS doesn't properly sync the cached information which leads to inconsistent results.
+
+Currently most of this work is still upheld because it would need some code refactoring in our bootloader, FreeLdr, as the implemented core registry healing routines makes bootloader size larger than what it should be which prevents the AMD64 build of ReactOS booting up due to a larger bootloader size.
```

Comment:
```suggestion
Currently most of this work is still being upheld because it makes our bootloader size larger than what it should be, preventing the AMD64 build of ReactOS booting up.
The implemented core registry healing routines require some refactoring in our bootloader, FreeLdr, before becoming mainstream.
```